src/coreComponents/linearAlgebra/CMakeLists.txt

```diff
@@ -57,6 +57,26 @@ if( ENABLE_MPI )
     list( APPEND dependencyList mpi )
 endif()
 
+if( ENABLE_SUITESPARSE )
+
+    list( APPEND linearAlgebra_headers
+          interfaces/direct/SuiteSparse.hpp )
+
+    list( APPEND linearAlgebra_sources
+          interfaces/direct/SuiteSparse.cpp )
+
+    list( APPEND dependencyList suitesparse )
+
+endif()
+
+list( APPEND linearAlgebra_headers
+      interfaces/direct/SuperLU_Dist.hpp )
+
+list( APPEND linearAlgebra_sources
+      interfaces/direct/SuperLU_Dist.cpp )
+
+list( APPEND dependencyList superlu_dist )
+
```

Comment:
Ok. If we decide that SuiteSparse is a required TPL, we can remove all these checks.
Be aware that, as of now, `thirdPartyLibs` assumes that the user can build them without SuiteSparse.
https://github.com/GEOSX/thirdPartyLibs/blob/509c390575b336828265b284176c7a0743005d6e/CMakeLists.txt#L45

Reply:
As a design principle, it would be good if we keep as many of the TPLs as "optional" as possible, even if they are enabled by default.  The more we require certain TPLs, the more trouble we will be in if they don't build on certain platforms.

Reply:
I agree with the design principle. However, with this PR we are requiring for almost all the integrated tests to use the direct solver provided by SuiteSparse as the linear solver. Making the majority on the integrated tests depending on an optional TPL seems a bit weird to me.

Reply:
> I also agree with the policy of minimizing the dependencies. Being flexible with TPL packages is essential, but here we heavily rely on this direct solver. I don't know if SuperLU can replace SuiteSparse for all the integratedTests.
> Instead, we could make SuperLU_Dist optional!

@af1990 We can have SuperLU_Dist optional. This choice has implication on _hypre_ though -- specifically the use of SuperLU_Dist as the coarsest level solver for boomerAMG:

https://github.com/GEOSX/GEOSX/blob/86c7c3a6e3ba54fde12cf58c658be87d49fa6076/src/coreComponents/linearAlgebra/interfaces/hypre/HyprePreconditioner.cpp#L166-L172

Reply:
Ok, maybe in a future PR we can think on this!